mediation-old/src/main/java/com/wso2telco/dep/mediator/util/HandlerUtils.java

```diff
@@ -77,10 +74,8 @@ public static void setAuthorizationHeader(MessageContext messageContext, Request
      *
      * @param messageContext the message context for which the gateway host is set
      */
-    public static void setGatewayHost(MessageContext messageContext){
-        FileReader fileReader = new FileReader();
-        String file = CarbonUtils.getCarbonConfigDirPath() + File.separator + FileNames.MEDIATOR_CONF_FILE.getFileName();
-        Map<String, String> mediatorConfMap = fileReader.readPropertyFile(file);
+    public static void setGatewayHost(MessageContext messageContext)throws Exception{
```

Comment:
Use this where applicable. See AmountChargeHandler.class

Reply:
Better not use this in the places which we have already fetched config map. ok to use if there is not fetched data available.